WordPress/wordpress-playground
# Support WordPress beta and RC versions

## Description

This pull request introduces support for WordPress beta and Release Candidate (RC) versions in the WordPress Playground.

Closes #500.
Closes #501.
Closes #306.

![CleanShot 2023-10-18 at 16 08 59@2x](https://github.com/WordPress/wordpress-playground/assets/205419/e3e7ac06-84e0-42ac-8b86-3152c554b49d)

## Changes

* The Dockerfile has been updated to automatically detect and keep the default theme for each WordPress release.
* Several updates to the build scripts and JSON files have been made to support these changes.
* The Blueprints no longer rely on a hardcoded list of supported WordPress versions. Instead, this PR introduces a new method where `remote.html` tells the client which versions it supports. This reduces coupling – the same client can work with different remotes.

## Testing instructions

1. Run `npx nx recompile-wordpress:all playground-remote`
2. Verify that the Playground successfully builds the latest 4 major WordPress versions and beta and nightly
3. Confirm the local Playground UI works well and allows switching between the different WordPress versions
